src/webapi/web_socket.rs

```diff
@@ -0,0 +1,197 @@
+use webcore::value::{Value, Reference, ConversionError};
+use webcore::try_from::{TryFrom, TryInto};
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::event::CloseEventCode;
+use webapi::blob::Blob;
+use webapi::array_buffer::ArrayBuffer;
+use webapi::typed_array::TypedArray;
+
+/// The WebSocket object provides the API for creating and managing a WebSocket connection to a
+/// server, as well as for sending and receiving data on the connection.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket)
```

Comment:
Please add a link to the specs (not in the doc comments; as a normal comment for us)

https://html.spec.whatwg.org/multipage/web-sockets.html#the-websocket-interface